### PR TITLE
Fixes #87 Messages turn red when deleted

### LINKED ISSUE #87: Add an indicator for deleted messages
In the current version, when a message is deleted, it is impossible to know if it was deleted as there is no indicator of it being deleted. It remains as is even after being refreshed. Only after WhatsApp redownloads its db does it update and show the "Restored message" tag. Until then, the message seems as if it was a normal message, it can be replied to, forwarded etc.

### PR DESCRIPTION
**Pull request by FieryRMS**
A message now turns red when someone deletes a message. Messages that fail to delete, and remains "virtually" deleted even after refreshing the page, turns red. Restored messages also turn red.

Known issues:
After a while, messages stop sending to certain group chats. Message are still received from the concerned group chat. Message can still be sent to other group chats and direct chats. Refreshing the page fixes the issue.
Could not trace the issue as it occurs randomly. Can't tell if the issue is caused by the base extension or my code or neither, but I observed it only after implementing my code.
A friend of mine who also uses my code did not complain of such problems.

### CODE REVIEW

**Comment by tomer8007** [comment]
But why do we need the `pseudomsgs` DB? Theoretically every message that gets deleted should be saved in the regular `msgs` DB after 4 seconds. Also, you have changes left from your own repo, like README.md and promotion.png

**Comment by FieryRMS** [comment]
After a message is deleted, it doesn't get deleted locally from the browser since we are blocking it. So even after refreshing, the site shows the message as if it wasn't deleted. Checking the message on the phone shows that it actually is. This is really confusing, hence the need for the indicator. The ```msgs``` db stores the would be id of the deleted message. Since it's not deleted, it doesn't "restore" it. Hence the need for a ```pseudomsgs``` db.

File: core/ui.js [hunk 1]
```diff
@@ -468,13 +481,16 @@ function onSafetyDelayEnabled()
...
     const messageText = messageNode.querySelector("." + UIClassNames.TEXT_WRAP_POSITION_CLASS + "." + UIClassNames.DELETED_MESSAGE_DIV_CLASS);
-    if (!messageText) return;
 
     if (messageNode.classList.contains("message-out")) return;
 
     const data_id = messageNode.getAttribute("data-id");
     const msgID = data_id.split("_")[2];
 
+    if (messageText || Allpsuedomsgskeys.has(msgID))
+    messageNode.setAttribute("data-deleted", "true");
```
Comment (tomer8007) [inline core/ui.js:491]: Please indent this line to the right

**Comment by tomer8007** [comment]
OK, I get it. And what happens if someone deletes a message while the page is already loaded? `Allpsuedomsgskeys` is updated only at `initialize`. Will it appear red immediately? 

Also, please just rename `pseudomsgs` to `pseudoMsgs` and `Allpsuedomsgskeys` to `psaeudoMsgsIDs` or so.

**Comment by FieryRMS** [comment]
Initially, I made the db thinking loading so many IDs into memory would slow down the extension. I suppose I didnt realize the issue would still happen eventually...

The message is turned red instantly when someone deletes a message on line [474 of ```interception.js```](https://github.com/tomer8007/whatsapp-web-incognito/pull/91/files#diff-0cfb6fd22f213741e13bf9bdb0651ebb51616cb26ea192f10910c0ae95ab6d55R474). A few lines later a event is fired to update ```psaeudoMsgsIDs``` in ```ui.js``` which is caught on [line 340](https://github.com/tomer8007/whatsapp-web-incognito/pull/91/files#diff-d86993c94dde63bbf4ae3c9b67c784381c84c7ee04fc05d7c71c7351019120ceR340).

**Comment by tomer8007** [comment]
Thanks

**Comment by tomer8007** [comment]
Thanks

**Pull request merged**